karapace/avro_dataclasses/schema.py

```diff
@@ -31,7 +31,7 @@ class EnumType(TypedDict):
 
 
 class MapType(TypedDict):
-    name: str
+    name: NotRequired[str]
```

Comment:
Is this used at all?

Reply:
I wrote this in a review too but apparently failed to press send:

---

I believe Map types should not have names: https://avro.apache.org/docs/1.10.2/spec.html#Maps. I believe a name only makes sense when the nested thing is something that can be referred to in one way or other, such as an Enum or nested Record type. Notably, it looks odd also that our ArrayType has a name.

I trust the backups suite excersises this well enough such that if we remove this and it passes, I'm comfortable.

Reply:
Yeah, this isn't used. I wasn't sure if removing was okay, but I'll clear it up then!